src/Catty/Parser&Serialializer/ProjectMigrator.swift

```diff
@@ -0,0 +1,114 @@
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class ProjectMigrator: ProjectMigratorProtocol {
+
+    private let correctVersionNumber: Float = 0.992
+    private var projectPath: String = ""
+    private var directoryPath: String = ""
+    var fileManager: CBFileManager!
+
+    enum FileSystem: String {
+        case automaticScreenShot = "automatic_screenshot.png"
+        case maualScreenShot = "manual_screenshot.png"
+        case screenShot = "screenshot.png"
+        case imagesDirectory = "images"
+        case soundDirectory = "sounds"
+    }
+
+    init(fileManager: CBFileManager) {
+        self.fileManager = fileManager
+    }
+
+    func migrateToScene(project: Project) throws {
+
+        guard let versionNumber = Float(project.header.catrobatLanguageVersion) else {
+            throw ProjectMigratorError.unknown(description: "Unable to convert version number to Float")
+        }
+
+        if versionNumber < self.correctVersionNumber {
+            throw ProjectMigratorError.invalidCatrobatLanguageVersion
+        }
+
+        self.projectPath = project.projectPath()
+
+        self.directoryPath = "\(projectPath + kLocalizedScene) 1"
+
+        if !fileManager.directoryExists(directoryPath) {
+            fileManager.createDirectory(directoryPath)
+        }
+
+        if fileManager.directoryExists(directoryPath) {
+
+            do {
+                try self.changeDirectory(for: FileSystem.automaticScreenShot.rawValue, isFile: true, isNecessary: true)
+                try self.changeDirectory(for: FileSystem.imagesDirectory.rawValue, isFile: false, isNecessary: true)
+                try self.changeDirectory(for: FileSystem.soundDirectory.rawValue, isFile: false, isNecessary: true)
+                try self.changeDirectory(for: FileSystem.maualScreenShot.rawValue, isFile: true, isNecessary: false)
+                try self.changeDirectory(for: FileSystem.screenShot.rawValue, isFile: true, isNecessary: false)
+            } catch {
+                throw error
+            }
+
+        } else {
+            throw ProjectMigratorError.unknown(description: "Unable to create \(kLocalizedScene) directory")
+        }
+
+    }
+
+    private func changeDirectory(for fileOrDirectory: String, isFile: Bool, isNecessary: Bool) throws {
```

Comment:
What does this method do?

Reply:
Hi @m-herold  I have changed the name of the function to `migrateToSceneDirectory(for projectResouce:, isFile:, shouldThrowIfRequired:)` to make itself explanatory.  This function moves each file/directory to the new directory `/Scene`.